components/provisioner/internal/provisioning/persistence/dbsession/read.go

```diff
@@ -511,3 +519,46 @@ func (r readSession) getOidcConfig(gardenerConfigID string) (model.OIDCConfig, d
 
 	return oidc, nil
 }
+
+func (r readSession) getDNSConfig(gardenerConfigID string) (*model.DNSConfig, dberrors.Error) {
+	var dnsConfigWithID struct {
+		model.DNSConfig
+		ID string `db:"id"`
+	}
+	var dnsProvidersPreSplit []struct {
+		model.DNSProvider
+		rawDomains string `db:"domains_include"`
+	}
+
+	err := r.session.
+		Select("*").
+		From("dns_config").
+		Where(dbr.Eq("gardener_config_id", gardenerConfigID)).
+		LoadOne(&dnsConfigWithID)
+
+	if errors.Is(err, dbr.ErrNotFound) {
+		return nil, nil
+	}
+	if err != nil {
+		return nil, dberrors.Internal("Failed to get DNS config: %s", err)
+	}
+
+	dnsConfig := dnsConfigWithID.DNSConfig
+
+	_, err = r.session.
+		Select("*").
```

Comment:
In both of these `Select`s (this one and the one above), I'd suggest explicitly listing the table columns instead of getting all of them with the `*` sign. Provisioner is not aware of the real schema with potential changes that are provided by Schema Migrator so its behavior could change if we leave it like this